proposals/NNNN-unlock-existentials-with-fixed-associated-types.md

```diff
@@ -0,0 +1,237 @@
+* Review Manager: TBD
+* Status: **Awaiting implementation**
+
+## Introduction
+
+Today, Swift protocols are divided into two categories: those that _can_ be used as fully-fledged types, and those that can only be used as generic constraints because they have `Self` or associated type requirements. However, in some cases where a protocol inherits and fixes the associated types of another protocol - that doesn't have `Self` requirements, this constraint seems rather unintuitive. This proposal aims to relax this needless contraint allowing such protocols to be used as Types.
+
+## Motivation
+
+Currently, any protocol that has associated type requirements is not allowed to be used as a Type. That's a sensible constraint; however, considering that the associated types of some protocols are known, this restriction can become quite frustrating: 
+
+```swift
+protocol User: Identifiable 
+    where ID == String {
+    
+    var username: String { get }
+    var displayName: String { get }
+}
+
+extension User {
+    var id: String {
+        username
+    }
+}
+```
+
+Many would rightfully assume that `User` could be used as a Type, since we specify that the `ID` asscoated type inherited from [`Identifiable`](https://github.com/apple/swift-evolution/blob/master/proposals/0261-identifiable.md) is `String`. 
+
+```swift
+let myUser: User
+// ❌ `User` is not
+// usable as a Type
+```
+> **_NOTE:_** One may point out that `myUser` could be bound to `some User`, which would have a similar effects. However, [opaque result types](https://github.com/apple/swift-evolution/blob/master/proposals/0244-opaque-result-types.md) are not Existentials. That is, the former allows for hiding type information from the user while the compiler internally retains the underlying type. On the contrary, the latter properly erases type information allowing storage of different types - that conform to a given protocol. What that means, is that should `myUser` be bound to `some User` then initialization would have to be performed at the declaration-site and mutation would be prohibited.
+
+This is a great inconvenience with not so elegant workarounds:
+
+```swift
+protocol AnyUser { ... } 
+// No `Identifiable` conformance
+
+let myUser: AnyUser
+// ✅
+
+
+protocol User: AnyUser, Identifiable 
+    where ID == String { 
+    
+    var id: String {
+        username
+    }
+}
+
+let myOtherUser: User 
+// ❌ Still an error as expected
+```
+
+These workarounds, besides producing boilerplate and confusing code, also produce confusing API designs leaving clients to wonder which of the two protocols should be used. In other cases, API authors may decide that adding `Identifiable` isn’t worth the added complexity and leave the `User` protocol with no such conformance whatsoever. 
+
+As a result, a straightforward conformance to simple - yet quite useful - protocols such as `Identifiable` and others becomes a hassle to implement.
```

Comment:
You probably mistook conformances for refinement or inheritance in these paragraphs.